cmd/util/cmd/leaders/cmd.go

```diff
@@ -0,0 +1,79 @@
+package leaders
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-go/cmd"
+	"github.com/onflow/flow-go/consensus/hotstuff/committees/leader"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol/inmem"
+)
+
+var (
+	flagStartView uint64
+	flagEndView   uint64
+)
+var Cmd = &cobra.Command{
+	Use:   "leaders",
+	Short: "Get leader selection for a view range.",
+	Long: `Get leader selection for a view range in the current epoch for a provided snapshot.
+ Expects a valid protocol state snapshot JSON to be piped into STDIN. Writes a JSON list of leaders for the given view range to STDOUT.`,
+	Run: run,
+}
+
+func init() {
+
+	Cmd.Flags().Uint64Var(&flagStartView, "start-view", 0, "the inclusive first view to get leader selection for")
+	Cmd.Flags().Uint64Var(&flagEndView, "end-view", 0, "the inclusive last view to get leader selection for")
+	cmd.MarkFlagRequired(Cmd, "start-view")
+	cmd.MarkFlagRequired(Cmd, "end-view")
+}
+
+func run(*cobra.Command, []string) {
+
+	var snapshot inmem.EncodableSnapshot
+	err := json.NewDecoder(os.Stdin).Decode(&snapshot)
```

Comment:
what's the usage? something like
```
cat snapshot.json | util ...
```

would it make sense to accept a path to the snapshot file as an argument instead?

Reply:
That is the usage. We were starting from some `jq` queries so the shortest path was something with the same input scheme. I don't mind adding the file IO in the command -- that is probably more consistent with the rest of the util commands.

Reply:
https://github.com/onflow/flow-go/pull/7160/commits/8195e7e34733bf0e042d4aba471f8c566839e0d2